websites/G/Google's Coding Competitions/presence.ts

```diff
@@ -0,0 +1,238 @@
+          break;
+        }
+
+        case "submissions": {
+          presenceData.details = `Viewing submissions for ${
+            document.querySelector(".rounds-header__container > p > b")
+              .textContent
+          }`;
+          presenceData.state =
+            document.querySelector(".headline-2").textContent;
+          break;
+        }
+      }
+      break;
+    }
+
+    case "codejam": {
+      presence = new Presence({ clientId: "848588299147935744" }); // Playing Google Code Jam
+      presenceData.largeImageKey = "codejam";
+
+      switch (location.pathname.split("/")[2]) {
+        case "about": {
+          presenceData.details = "Reading about Code Jam";
+          presenceData.smallImageKey = "reading";
+          break;
+        }
+
+        case "schedule": {
+          presenceData.details = "Viewing Schedule";
+          break;
+        }
+
+        case "faq": {
+          presenceData.details = "Reading FAQ";
+          presenceData.smallImageKey = "reading";
+          break;
+        }
+
+        case "rulesandterms": {
+          presenceData.details = "Reading Rules and Terms";
+          presenceData.smallImageKey = "reading";
+          break;
+        }
+
+        case "archive": {
+          if (!location.pathname.split("/")[3])
+            presenceData.details = "Viewing Competition archive";
+          else {
+            presenceData.details = "Viewing Competition archive";
+            presenceData.state =
+              document.querySelector(".headline-2").textContent;
+          }
+          break;
+        }
+
+        case "round": {
+          if (!location.pathname.split("/")[4]) {
+            presenceData.details = `Viewing ${
+              document.querySelector(".headline-2").textContent
+            }`;
+          } else {
+            const [round] = document
+                .querySelector(".headline-5")
+                .textContent.split("-"),
+              problem = document.querySelector(
+                "#problem-select-selected-text"
+              ).textContent;
+
+            if (!location.hash || location.hash.includes("#problem")) {
+              presenceData.details = `Solving: ${round}`;
+              presenceData.state = problem;
+            } else {
+              presenceData.details = `Reading Analysis: ${round}`;
+              presenceData.state = problem;
+              presenceData.smallImageKey = "reading";
+            }
+          }
+          break;
+        }
+
+        case "submissions": {
+          presenceData.details = `Viewing submissions for ${
+            document.querySelector(".rounds-header__container > p > b")
+              .textContent
+          }`;
+          presenceData.state =
+            document.querySelector(".headline-2").textContent;
+          break;
+        }
+      }
+      break;
+    }
+
+    case "hashcode": {
+      presence = new Presence({ clientId: "848588586524868701" }); // Playing Google Hash Code
```

Comment:
it's not good idea if you will update presence in UpdateData event :/

Reply:
sowwy